src/Git/Git.cpp

```diff
@@ -2291,6 +2291,14 @@ BOOL CGit::CheckMsysGitDir(BOOL bFallback)
 	}
 #endif
 
+	CString proxy = GetConfigValue(L"http.proxy");
+	if (!proxy.IsEmpty()) {
+		CString no_proxy = CRegString(L"Software\\TortoiseGit\\Git\\Servers\\global\\http-proxy-ignore", L"");
+		if (!no_proxy.IsEmpty()) {
+			m_Environment.SetEnv(L"NO_PROXY", no_proxy);

```

Comment:
With this we seem to have a proxy configuration in git config and also in the environment. I'd prefer to have it only in config ATM.

Reply:
The user can change the gitconfig file in the text editor.
Because of this, the settings in the gitconfig file are prioritized.

Reply:
Enrivonment variables have higher precedence, Therefore, "Because of this, the settings in the gitconfig file are prioritized." is wrong.

Reply:
I thought so too before. But source code of `libgit2` shows the opposite:
https://github.com/libgit2/libgit2/blob/b106834d70ae94d054c0127b35572010723fee4f/src/remote.c#L832-L846
Settings from the GIT config have a higher priority.

Reply:
libgit2 is not used for remote

Reply:
Ah, there is no "no proxy" config in Git.

Reply:
> Enrivonment variables have higher precedence
Look at this: https://github.com/curl/curl/blob/f3b7d72c57ed84189ce6e1ca3426e9681236ddf2/lib/url.c#L2437-L2478
The `curl` uses the `http_proxy` env setting only if there is no `-x` option in the startup parameters.

Reply:
Maybe a better approach is to set the environment variable at a different place.

Reply:
> Maybe a better approach is to set the environment variable at a different place.

Yeah. Calling `GetConfigValue` from the `CGit` constructor was not a good idea.
I had to make a special function `GetPreparedEnvironment` to change the environment.